services/cleanup/repository.py

```diff
@@ -0,0 +1,89 @@
+import logging
+from uuid import uuid4
+
+from django.db import transaction
+from shared.django_apps.codecov_auth.models import Owner
+from shared.django_apps.core.models import Repository
+
+from services.cleanup.cleanup import run_cleanup
+from services.cleanup.utils import CleanupSummary
+
+log = logging.getLogger(__name__)
+
+
+def cleanup_repo(repo_id: int) -> CleanupSummary:
+    cleanup_started, owner_id = start_repo_cleanup(repo_id)
+
+    if cleanup_started:
+        log.info("Started Repository cleanup", extra={"repo_id": repo_id})
+    else:
+        log.info("Continuing Repository cleanup", extra={"repo_id": repo_id})
+
+    repo_query = Repository.objects.filter(repoid=repo_id)
+    summary = run_cleanup(repo_query)
+    Owner.objects.filter(ownerid=owner_id).delete()
```

Comment:
just confirming that this is the deletion of the shadow owner we created in `start_repo_cleanup` ?

Reply:
yes it is. That function will detect based on the data whether the "shadow owner" was already created, and will return the ID of that one.